### DISCUSSION TITLE
Question: get current time within a job?

### DISCUSSION THREAD

**Discussion by bverbeken** [jobrunr/jobrunr#488] [Q&A] [answered]
I've got a pretty simple job, defined using a Java 8 lambda: 

```
scheduleRecurrently("job1", "*/5 * * * * *", () -> System.out.println("Job 1 " + LocalDateTime.now()));
```

I would expect this to print a new timestamp every 5 seconds. However, it does not. Instead, it prints the _same_ timestamp every 5 seconds. 

Clearly, I'm not using the system as intended and doing something wrong 😅 
However, I'm trying to understand why I see this behaviour; does it have to do with the fact that jobs get serialized? 🤔
And more importantly: if this is not the correct way, then how I can get access to the _current_ time in a job? For context: my job needs to hit a database and fetch & handle the records that were modified in the last X hours, hence my need for the current time. 

I've got a simple demo project here:  https://github.com/bverbeken/jobrunnr-with-time

Many thanks! 
Ben

**Comment by rdehuyss** [comment, marked as answer]
Hi Ben - no problem, it's a valid question!

Indeed, as the `LocalDateTime.now()` is serialised within the string of the sysout, it's fixed.

What I would do is just call a method without an argument and get the date in that method. So this would become:

```java
jobScheduler.scheduleRecurrently("job1", "*/5 * * * * *", this::fetchAndHandleNewRecords);

public void fetchAndHandleNewRecords() {
     System.out.println("Job 1 " + LocalDateTime.now()); // now the date will be updated on each run.
}
```

Does that make sense?

**Reply by bverbeken** [reply]
Absolutely! Thanks for the very quick reply 👍
